src/ui/public/agg_types/metrics/stdDeviation.js

```diff
@@ -7,26 +7,29 @@ export default function AggTypeMetricStandardDeviationProvider(Private) {
 
   var responseAggConfigProps = {
     valProp: function () {
-      var details = this.keyedDetails[this.key];
+      var details = this.keyedDetails(this.params.customLabel)[this.key];
       return details.valProp;
     },
     makeLabel: function () {
-      var details = this.keyedDetails[this.key];
+      var details = this.keyedDetails(this.params.customLabel)[this.key];
       return details.title + ' of ' + this.fieldDisplayName();
     },
-    keyedDetails: {
-      std_lower: {
-        valProp: ['std_deviation_bounds', 'lower'],
-        title: 'Lower Standard Deviation'
-      },
-      avg: {
-        valProp: 'avg',
-        title: 'Average'
-      },
-      std_upper: {
-        valProp: ['std_deviation_bounds', 'upper'],
-        title: 'Upper Standard Deviation'
-      }
+    keyedDetails: function (customLabel) {
+      const label = customLabel ? customLabel : 'Standard Deviation';
+      return {
+        std_lower: {
+          valProp: ['std_deviation_bounds', 'lower'],
+          title: 'Lower ' + label
+        },
+        avg: {
+          valProp: 'avg',
+          title: 'Average'
```

Comment:
Perhaps. I think it depends on what we expect people to put in for label. In the issue for this PR, #6407, @tbragin used "SD" as an example. I assume that's short for "standard deviation", which wouldn't make sense to include in the `avg` title, right?